### ISSUE TITLE
Use spaces consistently  

### ISSUE THREAD

**Issue by inverse** [mherrmann/gitignore_parser#16]
While Python allows you to use tabs or spaces the community often chooses spaces.

I noticed this repo contains a mixture with the majority tabs.

I would argue that spaces is more with the community and is actually part of PEP8.

See: https://www.python.org/dev/peps/pep-0008/

If you agree I can PR the changes to this and once the #15 is merged can add an additional linter to enforce on CI.

**Comment by mherrmann** [comment]
I'd be happy to merge a PR that changes to spaces. If the linter _only_ checks for spaces, that's fine. But please don't make it check for all PEP-8 checks. I think that's too much.

Thanks!

**Comment by inverse** [comment]
Okays once we get the #15 merged I'll tackle this one.

**Comment by bitranox** [comment]
@mherrmann , @inverse  - if You need help, let me know.
since we all talk german, we might meet on discord or smth like that.

I read Your PR´s and comments - here my opinion : 

- use black, then pep8 etc. shouldnt be an issue anymore.
- use flake8 - not pycodestyle
- use mypy - i dont see type annotations there, make PEP561 package
- I would prefer Travis CI over github workflows
- pytest over unittest i would say is ok - but avoid pytest plugins like : 
pytest-cov, pytest-pycodestyle, etc. - those plugins sometimes create more problems as they do good.

- tests and package should be somehow separated.

pytest is very good for auto discovery the tests ... 

settings for mypy, flake8 etc. should be slowly migrated to `pyproject.toml`

I am the author of `pizzacutter` and if You want I can convert the whole project very quickly into a pizzacutter project - 
the result (except the actual code) would be smth like that : 

https://github.com/bitranox/pct_python_default_test

After all that more administrative things are sorted out, here something I would actually want to change:

- put the matches into a class, so we can add `.gitignore`rules to the matches, 
also maybe add rules by `rule_from_pattern`

**Comment by inverse** [comment]
Hi @bitranox - I don't actually speak German (that well at least)

- I'm not well versed in the Python linting community aside a few of the tools like flake8 etc

- I prefer pytest too but @mherrmann isn't so keen so reverted my changed on #15 

> tests and package should be somehow separated.

I agree

**Comment by bitranox** [comment]
since I need a quick working solution, I made a fork ... 
@hermann - I really dont want to make a hostile takeover, but You seem to be busy with other things - so I made a quick fork.
I will be happy to pass it back to You after quickly moving forward and stabilisation period. 

@inverse - check out https://github.com/bitranox/igittigitt 
and make Your PRs against that repo, I need to move on quickly.
